crates/holochain/src/conductor/api/api_external/app_interface.rs

```diff
@@ -65,14 +65,16 @@ impl AppInterfaceApi for RealAppInterfaceApi {
                 self.conductor_handle.get_app_info(&app_id).await?,
             )),
             AppRequest::SignalSubscription(_subscription) => {
-                todo!("Signal pubsub not yet implemented")
+                unimplemented!("Signal pubsub currently unimplemented")
             }
             AppRequest::ZomeCallInvocation(request) => {
                 match self.conductor_handle.call_zome(*request).await? {
                     Ok(ZomeCallResponse::Ok(output)) => {
                         Ok(AppResponse::ZomeCallInvocation(Box::new(output)))
                     }
-                    Ok(ZomeCallResponse::Unauthorized) => Ok(AppResponse::ZomeCallUnauthorized),
+                    Ok(ZomeCallResponse::Unauthorized) => Ok(AppResponse::Error(
+                        ExternalApiWireError::ZomeCallUnauthorized("".into()),
```

Comment:
@Connoropolous could you just make the string include some info from the `request`, at least the zome and fn name?

We're using strings for now because we don't anticipate devs actually parsing these errors and doing something logical with them, but some basic info would be helpful for debugging.

Reply:
i did this, but now I blew up this PR a little bit with other stuffs. but this is fixed